src/script/interpreter.cpp

```diff
@@ -1951,32 +1978,46 @@ bool VerifyScript(const CScript &scriptSig,
             return set_error(serror, SCRIPT_ERR_SIG_PUSHONLY);
 
         // Restore stack.
-        swap(stack, stackCopy);
+        sm.setStack(stackCopy);
 
         // stack cannot be empty here, because if it was the
         // P2SH  HASH <> EQUAL  scriptPubKey would be evaluated with
         // an empty stack and the EvalScript above would return false.
-        assert(!stack.empty());
+        assert(!stackCopy.empty());
```

Comment:
no it's legacy.  I changed it to a DbgAssert that returns UNKNOWN_ERROR.  I don't think that this problem is "worth" a new error code because it really indicates a fundamental logic problem in the code above and so should be caught in testing.

Reply:
Had to change it back because DbgAssert uses LogPrint which does not exist in the library that interpreter.cpp is in, resulting in a compilation error on some architectures.